### ISSUE TITLE
Injection functionality should be implemented by a separate module from Test::Mockify

### ISSUE THREAD

**Issue by dbucky** [dbucky/Mockify#6]
Ideally, the injection capabilities of Mockify should not pollute the original implementation, but rather augment it. I like how Mockify was originally constructed to not interfere at all with the modules being mocked, and I think that should remain a core aspect of it.

However, I also am aware that it is not always possible to get the mock object into the system under test without modifying it for that express purpose. That is why the injection functionality was added. The purpose of this issue is to bring to light the inherent differences between the two approaches.

Let's say you have a `Dog` class with a `breed` function, which simply returns the breed specified when the Dog instance was created.
```perl
package Dog;
use strict;
use warnings;

sub new {
    my ($class, $breed) = @_;
    my $self = bless { breed => $breed }, $class;
    return $self;
}

sub breed {
    my $self = shift;
    return $self->{'breed'};
}

1;
```

Now let's mock this Dog using both approaches. The original usage of Mockify gives us something like the following. It creates an instance of the Dog class and then overrides the breed method on that instance only! This approach is useful when you have the ability to pass in an object instance to the system under test. In this case, the test object needs to be created ahead of time, and so the fact that a call to `Test::Mockify->new` creates the object instance on your behalf and passes along the necessary parameters to the constructor makes complete sense.
```perl
sub original_dog {
    my $dog = Dog->new('Dalmatian');
    is($dog->breed(), 'Dalmatian', "Original is a Dalmatian");

    my $mockify = Test::Mockify->new('Dog', ['Dalmatian']);
    $mockify->mock('breed')->whenAny()->thenReturn('Great Dane');
    my $mockObject = $mockify->getMockObject();

    is($mockObject->breed(), 'Great Dane', 'Mock is a Great Dane');
    is($dog->breed(), 'Dalmatian', "Original is a Dalmatian");

    ok(WasCalled($mockObject, 'breed'), 'Mocked breed method was called');
}
```

The static approach is somewhat simplified by comparison, since the mocked functionality is replacing the original implementation at the class level, and therefore applies to all object instances or static functions. The approach is useful for a system under test that creates a new instance of a class or calls a static function that you want to mock. In this case, the system under test handles setting up the instance, so Mockify should not be creating its own. The mocked class can be called directly in the test and provide the mocked functionality. The injector instance could provide the verification capabilities, as there is no mocked object.
```perl
sub static_dog {
    my $dog = Dog->new('Dalmatian');
    is($dog->breed(), 'Dalmatian', "Original is a Dalmatian");

    my $injector = Test::Mockify->new('Dog'); # should not call Dog->new() or need constructor parameters
    $injector->mockStatic('breed')->whenAny()->thenReturn('Great Dane');

    is(Dog->breed(), 'Great Dane', 'Mock is a Great Dane');
    is($dog->breed(), 'Great Dane', "Original is now a Great Dane");

    ok(WasCalled($injector, 'breed'), 'Mocked breed method was called');
}
```

To summarize:
- The injector should not create a mock object of the target.
- The injector's new method does not need parameters besides the target class.
- The injector should never call new on the target. It is replacing class functionality that applies to all object instances and static calls.
- The verification object will be the injector itself.

Because of these differences between the two approaches, having the injector be a separate module that inherits from `Test::Mockify` would allow it to diverge enough to achieve these goals.

**Comment by dbucky** [comment]
It would be difficult to determine which approach the developer intends to use at the point of `Test::Mockify->new`.

**Comment by dbucky** [comment]
I was able to address all of the requirements except "The verification object will be the injector itself." However, you can do:
```perl
$injector->getVerifier()
```
The `injector_issues` branch contains these changes.

**Comment by dbucky** [comment]
The remaining requirement is taken care of now: "The verification object will be the injector itself."
